### PR TITLE
Add back group chats messages

### PR DESCRIPTION
**Pull request by cammellos**
### Summary:

Adds group chats code but keeps flag off as further development will be needed.

* Moved to `extend-type` pattern for sending messages, similarly as we do for `receive`
* Removed unused protocol messages

### Added functionalities
* Create group chat
* Send messages
* Create group chat even if first message is not received

### Next steps

* Ensure `chat-id` cannot be manipulated by malicious users
* Sign and validate group membership changes
* Allow users to leave group chat
* Restore system messages

### Testing

light smoke tests

status: ready

### CODE REVIEW

File: src/status_im/chat/events.cljs [hunk 1]
```diff
@@ -128,22 +129,22 @@
...
  :create-new-group-chat-and-open
  [(re-frame/inject-cofx :random-id)]
  (fn [{:keys [db random-id] :as cofx} [_ group-name]]
-   (let [selected-contacts (:group/selected-contacts db)
-         chat-name         (if-not (string/blank? group-name)
-                             group-name
-                             (group-name-from-contacts selected-contacts
-                                                       (:contacts/contacts db)
-                                                       (:username db)))]
+   (let [my-pk             (:current-public-key db)
```
Comment (yenda) [inline src/status_im/chat/events.cljs]: we avoid any shorthand when in comes to name let's not make any exception

File: src/status_im/chat/models/group_chat.cljs [hunk 2]
```diff
@@ -30,7 +29,15 @@
       (seq removed-participants)
       (str admin-name " " (i18n/label :t/removed) " " (apply str (interpose ", " removed-participants-names))))))
 
-(defn handle-group-admin-update [{:keys [chat-name participants]} chat-id signature {:keys [now db random-id] :as cofx}]
+(defn handle-group-chat-create [{:keys [chat-name participants chat-id]} signature {:keys [now db random-id] :as cofx}]
+  (models.chat/add-group-chat
+   chat-id
```
Comment (yenda) [inline src/status_im/chat/models/group_chat.cljs]: can you put first argument on first line (sorry for nitpicking)

File: src/status_im/transport/core.cljs [hunk 3]
```diff
@@ -68,6 +68,6 @@
   account A messages without this."
   [{:keys [db]}]
   (let [{:transport/keys [chats discovery-filter]} db
-        chat-filters                               (mapv :filter (vals chats))
+        chat-filters                               (keep :filter (vals chats))
```
Comment (yenda) [inline src/status_im/transport/core.cljs:71]: :+1:

File: src/status_im/transport/message/transit.cljs [hunk 4]
```diff
@@ -55,35 +55,28 @@
...
-                             {v1.contact/NewContactKey           (NewContactKeyHandler.)
-                              v1.contact/ContactRequest          (ContactRequestHandler.)
-                              v1.contact/ContactRequestConfirmed (ContactRequestConfirmedHandler.)
-                              v1.contact/ContactUpdate           (ContactUpdateHandler.)
-                              v1.protocol/Message                (MessageHandler.)
-                              v1.protocol/MessagesSeen           (MessagesSeenHandler.)
-                              v1.group-chat/NewGroupKey          (NewGroupKeyHandler.)
-                              v1.group-chat/GroupAdminUpdate     (GroupAdminUpdateHandler.)
-                              v1.group-chat/GroupLeave           (GroupLeaveHandler.)}}))
+                             {v1.contact/NewContactKey            (NewContactKeyHandler.)
```
Comment (yenda) [inline src/status_im/transport/message/transit.cljs]: align all or nothing

**Review by rasom — APPROVED** [review]

File: src/status_im/transport/message/transit.cljs [hunk 5]
```diff
@@ -103,7 +96,9 @@
                               "c5" (fn [message-ids]
                                      (v1.protocol/MessagesSeen. message-ids))
                               "c6" (fn [[name profile-image address fcm-token]]
-                                     (v1.contact/ContactUpdate. name profile-image address fcm-token))}})) ; removed group chat handlers for https://github.com/status-im/status-react/issues/4506
+                                     (v1.contact/ContactUpdate. name profile-image address fcm-token))
+                              "g5" (fn [[chat-id chat-name admin participants signature message]]
```
Comment (yenda) [inline src/status_im/transport/message/transit.cljs:100]: what about g3 group leave for instance ?

File: src/status_im/transport/message/v1/protocol.cljs [hunk 6]
```diff
@@ -50,34 +73,6 @@
...
-                                             :payload payload}
-                                            whisper-opts)}}))
-
-;; TODO currently not used
-(defrecord Ack [message-ids]
-  message/StatusMessage
-  (send [this cofx chat-id])
-  (receive [this chat-id sig timestamp cofx]))
-
-(defrecord Seen [message-ids]
```
Comment (yenda) [inline src/status_im/transport/message/v1/protocol.cljs:76]: seen messages are completely removed ?

File: src/status_im/transport/message/v1/protocol.cljs [hunk 7]
```diff
@@ -38,6 +41,26 @@
...
                                        whisper-opts)}]}))
 
+(defn send-direct-message
+  "Sends the payload using to dst"
+  [{:keys [db] :as cofx} dst success-event payload]
+  (let [{:keys [current-public-key web3]} db]
+    {:shh/send-direct-message [{:web3 web3
+                                :success-event success-event
+                                :src     current-public-key
+                                :dst     dst
```
Comment (yenda) [inline src/status_im/transport/message/v1/protocol.cljs:51]: why stop use recipients ? what does dst stands for ?

File: src/status_im/transport/message/v1/protocol.cljs [hunk 8]
```diff
@@ -104,8 +112,13 @@
 (defrecord MessagesSeen [message-ids]
   message/StatusMessage
   (send [this chat-id cofx]
-    (send cofx
-          {:chat-id chat-id
-           :payload this}))
+    (if config/pfs-encryption-enabled?
+      (send-direct-message
+       chat-id
```
Comment (yenda) [inline src/status_im/transport/message/v1/protocol.cljs]: first argument on first line, cofx first, send-direct-message should use fx/defn

**Review by yenda — APPROVED** [review]

File: src/status_im/transport/message/transit.cljs (hunk 5 above)
Comment (cammellos) [reply src/status_im/transport/message/transit.cljs:100]: I think g3 was taken already, I had removed it, unlikely to be a problem but I just kept counting :)

File: src/status_im/transport/message/v1/protocol.cljs (hunk 6 above)
Comment (cammellos) [reply src/status_im/transport/message/v1/protocol.cljs:76]: I think we don't use it, we always use `MessagesSeen`?

File: src/status_im/transport/message/v1/protocol.cljs (hunk 7 above)
Comment (cammellos) [reply src/status_im/transport/message/v1/protocol.cljs:51]: destinations, I have aligned it with status-go, but can change, don't mind

File: src/status_im/transport/message/transit.cljs (hunk 5 above)
Comment (cammellos) [reply src/status_im/transport/message/transit.cljs:100]: ah sorry misunderstood, that's not implemented yet, so I won't be adding back for now

**Review by dmitryn — APPROVED** [review]

**Comment by lukaszfryc** [comment]
LGTM. I did not notice any regressions.

**Pull request merged**